src/status_list/mod.rs

```diff
@@ -0,0 +1,504 @@
+        }
+    }
+
+    pub fn decode_decompress(&self) -> Result<Vec<u8>, OAuthTSLError> {
+        let bytes = base64::engine::general_purpose::URL_SAFE_NO_PAD.decode(&self.status_list)?;
+        let mut d = ZlibDecoder::new(&bytes[..]);
+        let mut out = Vec::new();
+        d.read_to_end(&mut out)?;
+
+        Ok(out)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct StatusList {
+    #[serde(rename = "bits")]
+    pub status_size: Bits,
+    #[serde(rename = "lst")]
+    pub status_list: Vec<u8>,
+    // Aggregation URI is not implemented yet.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub aggregation_uri: Option<String>,
+}
+
+/// The default Status Size is 1 bit (VALID or INVALID) and no Aggregation Uri.
+/// The default length of the status list is 50 bytes, 400 bits.
+/// This is the recommended minimum size for a status list so to encourage "herd privacy" within a Status List.
+impl Default for StatusList {
+    fn default() -> Self {
+        StatusList {
+            status_size: Bits::One,
+            status_list: vec![0u8; 50],
+            aggregation_uri: None,
+        }
+    }
+}
+
+impl TryFrom<EncodedStatusList> for StatusList {
+    type Error = OAuthTSLError;
+
+    fn try_from(encoded_list: EncodedStatusList) -> Result<Self, Self::Error> {
+        let status_list = encoded_list.decode_decompress()?;
+        Ok(StatusList {
+            status_size: Bits::try_from(encoded_list.status_size)?,
+            status_list,
+            aggregation_uri: encoded_list.aggregation_uri,
+        })
+    }
+}
+
+impl StatusList {
+    /// Takes a Vec of Statuses, each represented by a u8 and packs them into a byte according to the status size.
+    /// A status size of 8 bits would therefore need no packing.
+    /// A status size of 1 bit would pack 8 statuses into a single byte.
+    pub fn pack_statuses_into_bytes(
+        &mut self,
+        statuses: Vec<StatusType>,
+    ) -> Result<Vec<u8>, OAuthTSLError> {
+        let status_size_usize = self.status_size as usize;
+        let statuses_per_byte = 8 / status_size_usize;
+        let mut packed: Vec<u8> = Vec::with_capacity(statuses.len().div_ceil(statuses_per_byte));
+        let mut byte: u8 = 0;
+        let mut bit_pos = 0;
+
+        for status in statuses.into_iter() {
+            if status as usize >= (1 << status_size_usize) {
+                return Err(OAuthTSLError::InvalidStatusType(status as u8));
+            }
+
+            byte |= (status as u8) << bit_pos;
+            bit_pos += status_size_usize;
+
+            if bit_pos >= 8 {
+                packed.push(byte);
+                byte = 0;
+                bit_pos = 0;
+            }
+        }
+
+        // Flush the last byte
+        if bit_pos > 0 {
+            packed.push(byte);
+        }
+
+        self.status_list = packed.clone();
+
+        Ok(packed)
+    }
+
+    pub fn get_index(&self, index: usize) -> Result<u8, OAuthTSLError> {
```

Comment:
`fn get_index(&self, index: usize)` looks a bit odd to me because `get_index` seems to imply that it is returning an `index` while it is also expecting an `index`. Would `get_status` perhaps a better name? And would it make sense to return `StatusType` instead of `u8`?